sql/src/main/java/io/crate/planner/operators/HashJoin.java

```diff
@@ -245,4 +261,29 @@ public long estimatedRowSize() {
     public <C, R> R accept(LogicalPlanVisitor<C, R> visitor, C context) {
         return visitor.visitHashJoin(this, context);
     }
+
+    private static class DistributionColumnExtractor extends SymbolVisitor<Void, Integer> {
+
+        private static final DistributionColumnExtractor INSTANCE = new DistributionColumnExtractor();
+
+        private static Integer getColumnIndex(Symbol symbol) {
+            return INSTANCE.process(symbol, null);
+        }
+
+        @Override
+        public Integer visitFunction(Function symbol, Void context) {
```

Comment:
Is this a valid scenario that we take the argument to a function to do the modulo distribution? I assume this is something like `inner join t2 on substr(t2.x, 0, 5) = t1.x` 

In which case modulo on `t2.x` instead of `substr..` and modulo on `t1.x` would lead to an incorrect distribution?